roomserver/internal/query/query.go

```diff
@@ -141,6 +142,18 @@ func (r *Queryer) QueryMissingAuthPrevEvents(
 	response.RoomExists = !info.IsStub
 	response.RoomVersion = info.RoomVersion
 
+	joined, err := r.DB.GetLocalServerInRoom(ctx, info.RoomNID)
+	if err != nil {
+		return fmt.Errorf("r.DB.GetLocalServerInRoom: %w", err)
+	}
+	response.RoomJoined = joined
+
+	// If we're not joined to the room then there's no point in hitting
+	// the database further to work out which events we're missing.
+	if !joined {
```

Comment:
This feels like a bad smell. `QueryMissingAuthPrevEvents` should do as it is told, regardless of the current room state if it has the data. Otherwise any caller to `QueryMissingAuthPrevEvents` needs to consider this which makes it hard to work with.

Reply:
We can leave it just to do the work and burn the effort, or we could add a request parameter to tell it whether to bother if we aren't joined to the room?

Reply:
It should be caller dependent really. Who is calling this API and when?